co-noir/ultrahonk/src/parse/crs.rs

```diff
@@ -0,0 +1,249 @@
+            num_g1_points: file.read_u32::<BigEndian>()?,
+            num_g2_points: file.read_u32::<BigEndian>()?,
+            start_from: file.read_u32::<BigEndian>()?,
+        })
+    }
+
+    fn write_manifest(filename: &str, manifest: &Manifest) -> Result<()> {
+        let mut file = File::create(filename)?;
+
+        // Here you need to call file.write_u32::<BigEndian>(value)? for each field in Manifest
+        file.write_u32::<BigEndian>(manifest.transcript_number)?;
+        file.write_u32::<BigEndian>(manifest.total_transcripts)?;
+        file.write_u32::<BigEndian>(manifest.total_g1_points)?;
+        file.write_u32::<BigEndian>(manifest.total_g2_points)?;
+        file.write_u32::<BigEndian>(manifest.num_g1_points)?;
+        file.write_u32::<BigEndian>(manifest.num_g2_points)?;
+        file.write_u32::<BigEndian>(manifest.start_from)?;
+
+        Ok(())
+    }
+    fn convert_endianness_inplace(buffer: &mut [u8]);
+}
+
+impl<P: Pairing> FileProcessor<P> for NewFileStructure<P> {
+    fn read_transcript_g1(
+        monomials: &mut [<P as Pairing>::G1Affine],
+        degree: usize,
+        path: &str,
+    ) -> Result<()> {
+        let g1_file_size = get_file_size(path)? as usize;
+        assert!(g1_file_size % 64 == 0); //g1_file_size >= num_points * 64 &&
+        let num_to_read = degree; //g1_file_size / 64;
+        let g1_buffer_size =
+            std::mem::size_of::<<P::G1 as CurveGroup>::BaseField>() * 2 * num_to_read;
+        let mut buffer = vec![0_u8; g1_buffer_size];
+
+        let file = File::open(path)?;
+        let mut file = file.take(g1_buffer_size as u64);
+        assert!(Path::new(&path).exists());
+        file.read_exact(&mut buffer[..])?;
+        // We must pass the size actually read to the second call, not the desired
+        // g1_buffer_size as the file may have been smaller than this.
+        let monomials = &mut monomials[0..];
+        Self::read_elements_from_buffer(monomials, &mut buffer);
+        Ok(())
+    }
+
+    fn read_transcript_g2(g2_x: &mut P::G2Affine, path: &str) -> Result<()> {
+        let g2_size = std::mem::size_of::<<P::G2 as CurveGroup>::BaseField>() * 2;
+
+        assert!(std::mem::size_of::<P::G2Affine>() >= g2_size);
+        let mut buffer = vec![0; g2_size];
+
+        let file = File::open(path)?;
+        let mut file = file.take(g2_size as u64);
+        file.read_exact(&mut buffer[..])?;
+        Self::convert_endianness_inplace(&mut buffer);
+        *g2_x = P::G2Affine::deserialize_uncompressed(&mut &buffer[..])
+            .map_err(|e| anyhow!("Failed to deserialize G2Affine from transcript file: {}", e))?;
+        Ok(())
+    }
+
+    // I don't think this is the nicest way, but the data is arranged slightly different as before
+    fn convert_endianness_inplace(buffer: &mut [u8]) {
+        for chunk in buffer.chunks_mut(32) {
+            if chunk.len() == 32 {
+                let mut reversed = [0u8; 32];
+                for i in 0..4 {
```

Comment:
Is this not just reversing the array completely?

Reply:
This is taken from barrustenberg. Updated this function to somehting simpler